### DISCUSSION TITLE
Could not load unit configuration (memfd_create failed)

### DISCUSSION THREAD

**Discussion by comebacktomorrow** [netbox-community/netbox-docker#1035] [Getting Help] [answered]
I'm trying to install netbox-docker on a Synology running docker 20.10.23 \ Docker Compose version v2.9.0-6413-g38f6acd \ containerd v1.6.15. I've tried installing through both Portainer, Synology container manager and straight up docker-compose and tried the last two versions and forks that people have used in their own tutorials (ie. modifying docker-compose-override) and I still end up with the same issue that I can't seem to get past. I realise this might be Synology issue, but I thought I'd post here just in case it's not.

At its most basic, my compose.override file just comprises of a custom port, a custom health check and environment for superuser and links to volumes.
```
⚙️ Applying configuration from /etc/unit/nginx-unit.json
2023/07/25 04:37:47 [alert] 38#38 *4 memfd_create(/unit.38.d0597756) failed (38: Function not implemented)
⚠️ Could no load Unit configuration
2023/07/25 04:37:47 [notice] 8#8 process 35 exited with code 1
2023/07/25 04:37:47 [notice] 8#8 process 38 exited with code 0
2023/07/25 04:37:47 [notice] 8#8 process 39 exited with code 0
```

Any advice as to what I can do to proceed past this? I'd really appreciate help! Thanks so much!

**Comment by kkthxbye-code** [comment, marked as answer]
Seems like your kernel doesn't support the `memfd_create` syscall. It was added in Linux 3.17/glibc 2.27, try to check what Linux kernel your Synology is running. If it's older than 3.17 and not possible to upgrade, I'm not sure if there is a great solution.

**Reply by comebacktomorrow** [reply]
Yeah wow. 3.10.108 unfortunately! Thanks for the simple answer!

**Reply by kkthxbye-code** [reply]
Yeah, that's very old. That kernel has been end of life for 6 years. Seems like Synology is really bad about upgrading their firmwares to include newer kernels, so you are probably out of luck sadly.

**Reply by comebacktomorrow** [reply]
Yep it seems that way! I am contemplating spinning up a VM instead .. we will see - keen to start exploring NetBox! Thanks again!
